azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java

```diff
@@ -1457,4 +1470,42 @@ public synchronized void handleEvent(final Event event) {
       }
     }
   }
+
+  /***
+   * Propagate properties (specified in {@code AZKABAN_EVENT_REPORTING_PROPERTIES_TO_PROPAGATE})
+   * to metadata for event reporting.
+   * @param metaData Metadata map to update with properties.
+   * @param inputProps Input properties for flow or job.
+   * @param nodeType Flow or job.
+   * @param nodeName Flow or job name.
+   * @param logger Logger from invoking class for log sanity.
+   */
+  public static void propagateMetadataFromProps(Map<String, String> metaData, Props inputProps, String nodeType,
```

Comment:
can we make the method package-private and annotate with @VisibleForTesting?

Reply:
Done